packages/lit-element/src/lib/updating-element.ts

```diff
@@ -669,37 +565,41 @@ export abstract class UpdatingElement extends HTMLElement {
       // in `update` this should not be possible (or an extreme corner case
       // that we'd like to discover).
       // mark state reflecting
-      this._updateState = this._updateState | STATE_IS_REFLECTING_TO_ATTRIBUTE;
+      this._reflectingProperty = name;
       if (attrValue == null) {
         this.removeAttribute(attr);
       } else {
         this.setAttribute(attr, attrValue as string);
       }
       // mark state not reflecting
-      this._updateState = this._updateState & ~STATE_IS_REFLECTING_TO_ATTRIBUTE;
+      this._reflectingProperty = null;
     }
   }
 
   private _attributeToProperty(name: string, value: string | null) {
-    // Use tracking info to avoid deserializing attribute value if it was
-    // just set from a property setter.
-    if (this._updateState & STATE_IS_REFLECTING_TO_ATTRIBUTE) {
-      return;
-    }
     const ctor = this.constructor as typeof UpdatingElement;
     // Note, hint this as an `AttributeMap` so closure clearly understands
     // the type; it has issues with tracking types through statics
     // tslint:disable-next-line:no-unnecessary-type-assertion
     const propName = (ctor._attributeToPropertyMap as AttributeMap).get(name);
-    if (propName !== undefined) {
+    // Use tracking info to avoid reflecting a property value to an attribute
+    // if it was just set because the attribute changed.
+    if (propName !== undefined && this._reflectingProperty !== propName) {
       const options = ctor.getPropertyOptions(propName);
       // mark state reflecting
-      this._updateState = this._updateState | STATE_IS_REFLECTING_TO_PROPERTY;
+      this._reflectingProperty = propName;
```

Comment:
Try to make variable declarations contiguous so that they can be minified into one declaration. Can you move this to below the declaration of `fromAttribute`?

Reply:
Done.